src/table-view/table-view.jsx

```diff
@@ -0,0 +1,92 @@
+import React, { Fragment } from 'react'
+import createReactClass from 'create-react-class'
+import { withStyles, Button } from '@material-ui/core'
+import _ from 'lodash'
+import Select from '../select'
+import Table from '../table'
+
+const TableView = createReactClass({
+  displayName: 'TableView',
+  propTypes: {},
+  getDefaultProps () {
+    return {
+      rows: [],
+      columns: []
+    }
+  },
+  getInitialState () {
+    return {
+      selectedValue: {},
+      selectArray: {},
+      rows: this.props.rows,
+      columns: this.props.columns,
+      selectType: ['genre', 'Rating']
+    }
+  },
+  componentDidMount () {
```

Comment:
why did you chose to make these calculations in `componentDidMount` ?

Reply:
I want to fetch the data as per the `selectType` from the `rows` JSON. We need to do it for the first time only. Once, the array is fetched, we don't need to call it again.